tensorly/tenalg/proximal.py

```diff
@@ -92,3 +95,159 @@ def procrustes(matrix):
     U, _, V = tl.partial_svd(matrix, n_eigenvecs=min(matrix.shape))
     return tl.dot(U, V)
 
+
+def hals_nnls(UtM, UtU, V=None, n_iter_max=500,tol=10e-8,
+                  sparsity_coefficient=None, normalize = False,nonzero_rows=False,exact=False):
+
+    """
+    Non Negative Least Squares (NNLS)
+
+    Computes an approximate solution of a nonnegative least
+    squares problem (NNLS) with an exact block-coordinate descent scheme.
+    M is m by n, U is m by r, V is r by n.
+    All matrices are nonnegative componentwise.
+
+    The NNLS unconstrained problem, as defined in [1], solve the following problem:
+
+            min_{V >= 0} ||M-UV||_F^2
```

Comment:
Either 
```rst 
following problem::

    # 4 space indent etc...
```

Generally, in the docstring, it should be readable rather than correct LaTex.
If it is helpful for comprehension, you can add more detail in the 
```rst
Notes
------
``` 
section, using the \`math\`:formula: direction or
```rst
.. math::

   formula
``` 
for blocks of math (see https://github.com/tensorly/tensorly/blob/main/tensorly/decomposition/robust_decomposition.py for an example of math in the doc, note the escaped backslash in the formula) 

Finally, a longer, more detailed and more technical explanation is perfect for the [user-guide](https://github.com/tensorly/tensorly/tree/main/doc/user_guide)!

Reply:
I have prepared the mathematical equations according to your second suggestion. I hope it is ok now.